utilities/opentelemetry_telemetry_metrics/lib/otel_telemetry_metrics.ex

```diff
@@ -0,0 +1,184 @@
+  @moduledoc """
+  `OtelTelemetryMetrics.start_link/1` creates OpenTelemetry Instruments for
+  `Telemetry.Metric` metrics and records to them when their corresponding
+  events are triggered.
+
+      metrics = [
+        last_value("vm.memory.binary", unit: :byte),
+        counter("vm.memory.total"),
+        counter("db.query.duration", tags: [:table, :operation]),
+        summary("http.request.response_time",
+          tag_values: fn
+            %{foo: :bar} -> %{bar: :baz}
+          end,
+          tags: [:bar],
+          drop: fn metadata ->
+            metadata[:boom] == :pow
+          end
+        ),
+        sum("telemetry.event_size.metadata",
+          measurement: &__MODULE__.metadata_measurement/2
+        ),
+        distribution("phoenix.endpoint.stop.duration",
+          measurement: &__MODULE__.measurement/1
+        )
+      ]
+
+      {:ok, _} = OtelTelemetryMetrics.start_link([metrics: metrics])
+
+  Then either in your Application code or a dependency execute `telemetry`
+  events conataining the measurements. For example, an event that will result
+  in the metrics `vm.memory.total` and `vm.memory.binary` being recorded to:
+
+      :telemetry.execute([:vm, :memory], %{binary: 100, total: 200}, %{})
+
+  OpenTelemetry does not support a `summary` type metric, the `summary`
+  `http.request.response_time` is recorded as a single bucket histogram.
+
+  In `Telemetry.Metrics` the `counter` type refers to counting the number of
+  times an event is triggered, this is represented as a `sum` in OpenTelemetry
+  and when recording the value is sent as a `1` every time.
+
+  Metrics of type `last_value` are ignored because `last_value` is not yet an
+  aggregation supported on synchronous instruments in Erlang/Elixir
+  OpenTelemetry. When it is added to the SDK this library will be updated to
+  no longer ignore metrics of this type.
+  """
+
+  require Logger
+  use GenServer
+
+  @doc """
+  """
+  def start_link(options) do
+    GenServer.start_link(__MODULE__, options)
+  end
+
+  @impl true
+  def init(options) do
+    Process.flag(:trap_exit, true)
+
+    meter = options[:meter] || :opentelemetry_experimental.get_meter()
+    metrics = options[:metrics] || []
+
+    handler_ids = create_instruments_and_attach(meter, metrics)
+
+    {:ok, %{handler_ids: handler_ids}}
+  end
+
+  @impl true
+  def terminate(_, %{handler_ids: handler_ids}) do
+    detach(handler_ids)
+    :ok
+  end
+
+  defp create_instruments_and_attach(meter, metrics) do
+    metrics_by_event = Enum.group_by(metrics, & &1.event_name)
+
+    for {event_name, metrics} <- metrics_by_event do
+      instruments = create_instruments(meter, metrics)
+
+      attach(meter, event_name, instruments)
+    end
+  end
+
+  defp create_instruments(meter, metrics) do
+    for metric <- metrics,
+      instrument = create_instrument(metric, meter, %{unit: unit(metric.unit)}),
+      instrument != nil, into: %{} do
+        {metric, instrument}
+    end
```

Comment:
Comprehensions automatically filter nil values, so you can skip `instrument != nil`.

```suggestion
    for metric <- metrics,
        instrument = create_instrument(metric, meter, %{unit: unit(metric.unit)}),
        into: %{} do
        {metric, instrument}
    end
```

Reply:
umm.. I don't think it's true..

I just tried:
```elixir
iex> for x <- [1, nil, 2, 3], do: x
[1, nil, 2, 3]
```
neither it filters `nil` when passed `into: %{}`
```elixir
iex> for x <- [1, nil, 2, 3], into: %{}, do: {x, x}
%{1 => 1, 2 => 2, 3 => 3, nil => nil}
```

Reply:
Comprehension filters automatically do:

```
iex> for x <- [1, nil, 2, 3], x, into: %{}, do: {x, x}
```

Reply:
Oh, I see, thank you!

Reply:
Thanks!